lib/IMAP/IMAPClientFactory.php

```diff
@@ -117,8 +117,16 @@ public function getClient(Account $account, bool $useCache = true): Horde_Imap_C
 				'backend' => $this->hordeCacheFactory->newCache($account),
 			];
 		}
-		if ($this->config->getSystemValue('debug', false)) {
-			$params['debug'] = $this->config->getSystemValue('datadirectory') . '/horde_imap.log';
+		$debug = array_merge(
```

Comment:
If we decide to keep the protocol option, then this code should go into a small service or into the account object. 

```mermaid
classDiagram
    class DebugHelper
    DebugHelper : +debugImap(account) bool
    DebugHelper : +debugSmtp(account) bool
    DebugHelper : +debugSieve(account) bool
    DebugHelper : +fileNameImap(account) string
    DebugHelper : +fileNameSmtp(account) string
    DebugHelper : +fileNameSieve(account) string
```

Reply:
Sure, that can be done. Let see what the rest think.